Remove hard coded curses library name

While building NEURON on OpenSuse I see below error:

```
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: cannot find -lcurses
collect2: error: ld returned 1 exit status
make[2]: *** [src/nrniv/CMakeFiles/nrniv_lib.dir/build.make:10948: lib/libnrniv.so] Error 1
make[2]: Leaving directory '/nrn/build'
make[1]: *** [CMakeFiles/Makefile2:233: src/nrniv/CMakeFiles/nrniv_lib.dir/all] Error 2
make[1]: Leaving directory '/nrn/build'
make: *** [Makefile:130: all] Error 2
```

This is because, we have hard coded library name in src/readline/CMakeLists.txt

```
# =============================================================================
# readline library
# =============================================================================
add_library(readline STATIC ${READLINE_SOURCE_FILES})
target_link_libraries(readline curses)
```

But CMake find library as:

```
-- Found Curses: /usr/lib64/libncurses.so
```

I installed curses library with:

```
zyper search ncurses
```

--- a/src/readline/CMakeLists.txt
+++ b/src/readline/CMakeLists.txt
@@ -13,7 +13,7 @@ find_package(Curses REQUIRED QUIET)
 # readline library
 # =============================================================================
 add_library(readline STATIC ${READLINE_SOURCE_FILES})
-target_link_libraries(readline curses)
+target_link_libraries(readline ${CURSES_LIBRARIES})
 set_property(TARGET readline PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 # =============================================================================

